File: src/main/java/br/com/eteg/opbp/entities/Account.java
package br.com.eteg.opbp.entities;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
public class Account {
  private @Id @GeneratedValue Integer id;
  private String username;
  private @JsonIgnore String password;
  private String name;
  private String email;

  public Account() {
  }

  public Account(String username, String password, String name, String email) {
    this.username = username;
    this.password = password;
    this.name = name;
    this.email = email;
  }

  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getUsername() {
    return username;
  }

  public void setUsername(String username) {
    this.username = username;
  }

  public String getPassword() {
    return password;
  }

  public void setPassword(String password) {
    this.password = password;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getEmail() {
    return email;
  }

  public void setEmail(String email) {
    this.email = email;
  }
}

File: src/main/kotlin/br/com/eteg/opbp/entities/Account.kt
package br.com.eteg.opbp.entities

import br.com.eteg.opbp.entities.base.BaseEntity
import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "opbp-accounts")
class Account(
        @Id
        override var id: String? = null,

        var username: String? = null,

        @JsonIgnore
        var password: String? = null,

        var name: String? = null,

        var email: String? = null
) : BaseEntity<String>()